frontend/web/components/GitHubResourcesSelect.tsx

```diff
@@ -1,147 +1,61 @@
-        if (loadingCombinedData) {
-          ;(
-            infiniteLoaderRef.current as InfiniteLoader
-          ).resetloadMoreItemsCache()
-        }
-      }
-    }
-    hasMountedRef.current = true
-  }, [searchText, loadingCombinedData])
-
-  return isLoading ? (
-    <div>Loading...</div>
-  ) : !itemCount ? (
-    <div>No results found</div>
-  ) : (
-    <InfiniteLoader
-      isItemLoaded={isItemLoaded}
-      itemCount={itemCount}
-      loadMoreItems={moreItems}
-      threshold={40}
-    >
-      {({ onItemsRendered, ref }) => (
-        <FixedSizeList
-          {...rest}
-          height={Math.min(
-            itemCount * itemSize + 20,
-            selectProps.maxMenuHeight,
-          )}
-          itemCount={itemCount}
-          itemSize={itemSize}
-          onItemsRendered={onItemsRendered}
-          ref={ref}
-          width={width}
-        >
-          {({ index, isScrolling, style, ...rest }) => {
-            const child = childrenArray[index]
-            return (
-              <div {...rest} style={style}>
-                {isItemLoaded(index) ? (
-                  child
-                ) : (
-                  <div className='text-center'>
-                    <Loader />
-                  </div>
-                )}
-              </div>
-            )
-          }}
-        </FixedSizeList>
-      )}
-    </InfiniteLoader>
-  )
-}
-
-const CustomControl = ({
-  children,
-  ...props
-}: {
-  children: React.ReactNode
-}) => {
-  const { refresh } = useGitHubResourceSelectProvider()
-  return (
-    <components.Control {...props}>
-      {children}
-      <Button style={{ marginLeft: 'auto' }} theme='icon' onClick={refresh}>
-        <Icon name='refresh' />
-      </Button>
-    </components.Control>
-  )
-}
+import useInfiniteScroll from 'common/useInfiniteScroll'
+import { Req } from 'common/types/requests'
+import { useGetGithubResourcesQuery } from 'common/services/useGithub'
+import Constants from 'common/constants'
+import MyRepositoriesSelect from './MyRepositoriesSelect'
 
 export type GitHubResourcesSelectType = {
   onChange: (value: string) => void
   lastSavedResource: string | undefined
+  linkedExternalResources: ExternalResource[] | undefined
+  orgId: string
+  githubId: string
+  resourceType: string
+  setResourceType: (value: string) => void
 }
 
 type GitHubResourcesValueType = {
   value: string
 }
-
 const GitHubResourcesSelect: FC<GitHubResourcesSelectType> = ({
+  githubId,
   lastSavedResource,
   onChange,
+  orgId,
+  resourceType,
+  setResourceType,
 }) => {
-  const { githubResources, isFetching, isLoading, searchItems } =
-    useGitHubResourceSelectProvider()
+  const githubTypes = Object.values(Constants.resourceTypes).filter(
+    (v) => v.type === 'GITHUB',
+  )
+  const [repo, setRepo] = useState('')
+  const { repoName, repoOwner } = useMemo(() => {
+    if (!repo) {
+      return {}
+    }
+    const [repoOwner, repoName] = repo.split('/')
```

Comment:
There seems to be an error here. The `repo` is structured as repo-name/repo-owner, which is causing an issue with the API.

<img width="525" alt="Screenshot 2024-08-15 at 9 55 21 AM" src="https://github.com/user-attachments/assets/c0958643-027a-4cca-b768-b304b14d4777">

Reply:
ah, that's what I get for testing with flagsmith/flagsmith